Sources/X509/Verifier/OneOfPolicyBuilder.swift

```diff
@@ -0,0 +1,207 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftASN1
+
+/// Provides a result-builder style DSL for constructing a ``VerifierPolicy`` in which one of the specified policies must match
+///
+/// This DSL allows us to construct dynamic ``VerifierPolicy`` at runtime without using type erasure.
+/// The resulting ``VerifierPolicy`` will use the listed policy in the order of declaration to check if a chain meets any one policy
+@resultBuilder
+public struct OneOfPolicyBuilder {}
+
+extension OneOfPolicyBuilder {
+    @inlinable
+    public static func buildLimitedAvailability<Policy: VerifierPolicy>(_ component: Policy) -> Policy {
+        component
+    }
+}
+
+// MARK: empty policy
+extension OneOfPolicyBuilder {
+    @usableFromInline
+    struct Empty: VerifierPolicy {
+        @inlinable
+        var verifyingCriticalExtensions: [SwiftASN1.ASN1ObjectIdentifier] { [] }
+
+        @inlinable
+        init() {}
+
+        @inlinable
+        mutating func chainMeetsPolicyRequirements(chain: UnverifiedCertificateChain) async -> PolicyEvaluationResult {
+            .failsToMeetPolicy(reason: "No policies specified in OneOfPolicies block")
+        }
+    }
+
+    @inlinable
+    public static func buildBlock() -> some VerifierPolicy {
+        Empty()
+    }
+}
```

Comment:
We don't need to support empty OneOfPolicies. We can remove this.

Reply:
We could also keep it and mark it unavailable.
With that being said, I'm actually not feeling to strong here as it has the right behaviour so we can keep it.
Ideally we would also report an error for only one policy because it would be useless.However, it also isn't unsafe and we don't run into any unexpected behavior. It does what it looks like so it's fine that users can write this code after all.

Reply:
talked to @Lukasa offline and we prefer to keep it as is, because there is no harm in having an empty policy